staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go

```diff
@@ -35,6 +36,12 @@ func ErrBadArgType(expected, actual string) error {
 	return fmt.Errorf("expected a %s, but received a %s", expected, actual)
 }
 
+func ErrBadArgTypeAssertion(expected, actual interface{}) error {
```

Comment:
There are some usage left. e.g. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go#L571.
WDYT leave them as is?